internal/templates/readonlyrootfs/template.go

```diff
@@ -0,0 +1,40 @@
+package readonlyrootfs
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/extract"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+)
+
+func init() {
+	templates.Register(check.Template{
+		Name:        "read-only-root-fs",
+		Description: "Flag containers without read-only root file systems",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters: nil,
+		Instantiate: func(_ map[string]string) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				podSpec, found := extract.PodSpec(object.K8sObject)
+				if !found {
+					return nil
+				}
+				var results []diagnostic.Diagnostic
+				for i := range podSpec.Containers {
```

Comment:
Why not range with _, container?

Reply:
IIRC linter complaining about large object for loop iteration variable. Though since you do `container := podSpec.Containers[i]`, really nothing is gained here. So either change the assignment to `:= &podSpec.Containers[i]`, or add a nolint directive and apply gorman's suggestion

Reply:
Just removed this specific lint check. Not convinced it added enough value to justify the clunkiness.